src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -163,6 +186,9 @@ public ProvisionConfiguration updateIdentityConfiguration(ProvisionContext provi
         } catch (CrtRuntimeException | InterruptedException ex) {
             logger.atError().setCause(ex).log("Exception encountered while getting device identity information");
             throw ex;
+        } catch (IOException ex) {
+            logger.atError().setCause(ex).log("CSR file couldn't be read");
+            return null;
```

Comment:
returning null is not valid in this method. The exception must be thrown up to the caller. Wrap the IOException as a RuntimeException, such as `throw new RuntimeException(ex)`